libr/anal/sign.c

```diff
@@ -774,6 +766,120 @@ R_API bool r_sign_add_anal(RAnal *a, const char *name, ut64 size, const ut8 *byt
 	return retval;
 }
 
+static RSignGraph *r_sign_fcn_graph(RAnalFunction *fcn) {
+	r_return_val_if_fail (fcn, false);
+	RSignGraph *graph = R_NEW0 (RSignGraph);
+	if (graph) {
+		graph->cc = r_anal_function_complexity (fcn),
+		graph->nbbs = r_list_length (fcn->bbs);
+		graph->edges = r_anal_function_count_edges (fcn, &graph->ebbs);
+		graph->bbsum = r_anal_function_realsize (fcn);
+	}
+	return graph;
+}
+
+static RSignBytes *r_sign_fcn_bytes(RAnal *a, RAnalFunction *fcn) {
+	r_return_val_if_fail (a && fcn, false);
+	RCore *core = a->coreb.core;
+	int maxsz = a->coreb.cfggeti (core, "zign.maxsz");
+
+	// get function real size
+	int fcnlen = 0;
+	RListIter *iter;
+	RAnalBlock *bb = NULL;
+	r_list_foreach (fcn->bbs, iter, bb) {
+		fcnlen += bb->size;
+		if (fcnlen > maxsz) {
+			break;
+		}
+	}
+
+	int size = R_MIN (core->io->addrbytes * fcnlen, maxsz);
+
+	RSignBytes *sig = R_NEW0 (RSignBytes);
+	if (!sig) {
+		goto bytes_failed;
+	}
+	sig->size = size;
```

Comment:
no this is not correct either, because you are counting bbs that are not lineart and then reading linearly. i think the way to go for this would be to create a map with a mask for all the linear bytes of the function starting from the entry. so if a function starts with a jmpback it wont be detected but if there's data in the middle of the code it will be a valid match. can you redo this to follow this logic?

Reply:
Sorry, I misunderstood earlier. To make sure I got it now I made a simple example.
```
_jumpback:
    inc rbx
    jmp _next

 ; not part of function
_ignoreme:
    inc r10

;; entry point of function
_entry:           ;<-- start matching signature
    inc rax       ;   |                        
    jmp _jumpback ;   |                              
                  ;   |             
_next:            ;   |                   
    inc rdx       ;   |                        
    jmp _end      ;<--+ end matching signature                       


;; hole in function
_junk:
    inc rbp

_end:
    ret
```

I can correct the size to implement this logic.
Let me know if I am still misunderstanding and tell me which blocks in the above example should be included.